client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -345,4 +358,77 @@ public void close() {
             this.connectionPool.close();
         }
     }
+
+    @Override
+    public CompletableFuture<Long> getDistanceBetweenTwoStreamCuts(Stream stream, StreamCut fromStreamCut, StreamCut toStreamCut) {
+        Preconditions.checkNotNull(stream, "Ensure to pass a valid stream");
+        Preconditions.checkNotNull(fromStreamCut, "fromStreamCut");
+        Preconditions.checkNotNull(toStreamCut, "toStreamCut");
+        final CompletableFuture<StreamSegmentSuccessors> unread;
+        final Map<Segment, Long> endPositions;
+        if (!fromStreamCut.equals(StreamCut.UNBOUNDED) && toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            Preconditions.checkArgument(stream.getScopedName().equals(fromStreamCut.asImpl().getStream().getScopedName()), "Ensure that the stream passed is same as that of the input streamcut");
+            StreamCut endSC = getTailStreamCut(stream);
+            unread = controller.getSegments(fromStreamCut, endSC);
+            endPositions = endSC.asImpl().getPositions();
+        } else if (fromStreamCut.equals(StreamCut.UNBOUNDED) && !toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            Preconditions.checkArgument(stream.getScopedName().equals(toStreamCut.asImpl().getStream().getScopedName()), "Ensure that the stream passed is same as that of the input streamcut");
+            StreamCut startSC = getHeadStreamCut(stream);
+            unread = controller.getSegments(startSC, toStreamCut);
+            endPositions = toStreamCut.asImpl().getPositions();
+        } else if (fromStreamCut.equals(StreamCut.UNBOUNDED) && toStreamCut.equals(StreamCut.UNBOUNDED)) {
+            StreamCut startSC = getHeadStreamCut(stream);
+            StreamCut endSC = getTailStreamCut(stream);
+            unread = controller.getSegments(startSC, endSC);
+            endPositions = endSC.asImpl().getPositions();
```

Comment:
This is really just doing what is done in both of the previous conditions. So to simplify the logic this flow could made into 2 ifs with some code after  rather than 4 if/else blocks.

Reply:
Incorporated as suggested